src/components/RegisterBox/RegisterBox.js

```diff
@@ -8,34 +8,47 @@ import { connect } from 'react-redux';
 import './RegisterBox.css';
 
 class RegisterBox extends PureComponent {
-
   state = {
     username: '',
     email: '',
     password: '',
     password2: '',
     errors: {},
-  }
+  };
 
   handleRegister = e => {
+    const {
+      username,
+      email,
+      password,
+      password2,
+    } = this.state;
+
     e.preventDefault();
 
     const user = {
-      username: this.state.username,
-      email: this.state.email,
-      password: this.state.password,
-      password2: this.state.password2
-    }
+      username,
+      email,
+      password,
+      password2,
+    };
 
     this.props.registerUser(user, this.props.history);
-  }
+  };
 
-  onTextChange = e => {
-    this.setState({ [e.target.name]: e.target.value });
-  }
+  onTextChange = ({ target: { name, value } }) => {
+    this.setState({ [name]: value });
+  };
 
   render() {
-    const { errors } = this.props;
+    const {
+      errors: {
+        email: emailErrors,
```

Comment:
I'm also a big fan of keeping values held by other values (e.g. `{ errors: { email: [] } }`) as generic as possible, as you've done here. Looking at it, you know `errors.email` would the the email errors. However when you begin using that value elsewhere among other components/variables, using the value `email` here wouldn't be indicative of what the variable held. Since here these values are used often and destructuring is a best practice, I've renamed them to show what they actually contain. Since you don't have the context of `errors.email`, you now see `emailErrorrs` and know what this means.

Reply:
thanks for that explanation!